jetstream/external_config.py

```diff
@@ -39,6 +46,99 @@ def validate(self, experiment: jetstream.experimenter.Experiment) -> None:
         Analysis("no project", "no dataset", conf).validate()
 
 
+def check_all_keys_lowercase(config: MutableMapping[str, Any]) -> Dict[Any, Any]:
+    """
+    function to recursively check that configuration keys are lowercase.
+
+    Returns a dictionary with a dict key path to key which is invalid.
+
+    """
+
+    invalid_keys = dict()
+    for key, value in config.items():
+        if isinstance(value, dict):
+            result = check_all_keys_lowercase(value)
+            if result:
+                invalid_keys[key] = result
+
+        if key.lower() != key:
+            invalid_keys[key] = {key: "INVALID_CONFIG_SETTING"}
```

Comment:
I guess we could simplify this to just collect the keys and remove `INVALID_CONFIG_SETTING`?